src/Uno.UI/ts/WindowManager.ts

```diff
@@ -1479,24 +1480,35 @@ namespace Uno.UI {
 					const imgElement = element as HTMLImageElement;
 					return [imgElement.naturalWidth, imgElement.naturalHeight];
 				}
-				else if (element instanceof HTMLInputElement) {
-					const inputElement = element as HTMLInputElement;
+				else if (element instanceof HTMLInputElement || element instanceof HTMLTextAreaElement) {
+					const inputElement = element;
 
 					cleanupUnconnectedRoot(this.containerElement);
 
 					// Create a temporary element that will contain the input's content
 					var textOnlyElement = document.createElement("p") as HTMLParagraphElement;
 					textOnlyElement.style.cssText = updatedStyleString;
 					textOnlyElement.innerText = inputElement.value;
+					textOnlyElement.className = elementClasses;
 
 					unconnectedRoot = textOnlyElement;
 					this.containerElement.appendChild(unconnectedRoot);
 
 					var textSize = this.measureElement(textOnlyElement);
 					var inputSize = this.measureElement(element);
 
-					// Take the width of the inner text, but keep the height of the input element.
-					return [textSize[0], inputSize[1]];
+					// Take the width of the inner text. For HTMLInputs, keep the height of the input element as they are always one line.
+					// For HTMLTextArea, take the height of the ParagraphElement.
+					const width = Math.min(textSize[0], maxWidth);
+					var height;
+					if (element instanceof HTMLTextAreaElement) {
+						height = Math.max(textSize[1] + 1, inputSize[1]);
```

Comment:
why do you add 1?

Reply:
same reason we add 1 to the width in measureelement, to account for flooring